### PR TITLE
Feature/actions as obs

### PR DESCRIPTION
**Pull request by michele-milesi**
## Summary

Describe the purpose of the pull request, including:

- Add `ActionsAsObservationWrapper`. The wrapper takes in input the number of the last actions to add to the observations and the dilation to use. It works as the `FrameStack` wrapper.
- Diambra environments do not use this wrapper, the library already provides a wrapper for adding the last `n` actions to the observations.

## Type of Change

Please select the one relevant option below:

- New feature (non-breaking change that adds functionality)

## Checklist

Please confirm that the following tasks have been completed:

- [x] I have tested my changes locally and they work as expected. (Please describe the tests you performed.)
- [x] I have added unit tests for my changes, or updated existing tests if necessary.
- [x] I have updated the documentation, if applicable.
- [x] I have installed pre-commit and run locally for my code changes.

### CODE REVIEW

File: sheeprl/envs/wrappers.py [hunk 1]
```diff
@@ -251,3 +253,65 @@ def render(self) -> Optional[Union[RenderFrame, List[RenderFrame]]]:
...
+    def step(self, action: Any) -> Tuple[Any | SupportsFloat | bool | Dict[str, Any]]:
+        self._actions.append(action)
+        obs, reward, done, truncated, info = super().step(action)
+        obs["action_stack"] = self._get_actions_stack()
+        return obs, reward, done, truncated, info
+
+    def reset(self, *, seed: int | None = None, options: Dict[str, Any] | None = None) -> Tuple[Any | Dict[str, Any]]:
+        obs, info = super().reset(seed=seed, options=options)
+        self._actions.clear()
+        if self._is_multidiscrete or self._is_continuous:
```
[the remark this answers is not in the document]
Comment (michele-milesi) [reply sheeprl/envs/wrappers.py:299]: Yeah, I had thought of adding a wrapper that adds the NOOP action (one wrapper for each environment) and use it when resetting. That was the only alternative option I could think of.
Do you have any other ideas?

[the remark this answers is not in the document]
Comment (michele-milesi) [reply sheeprl/envs/wrappers.py:299]: LGTM

File: sheeprl/utils/env.py [hunk 2]
```diff
@@ -207,6 +208,9 @@ def transform_obs(obs: Dict[str, Any]):
                 )
             env = FrameStack(env, cfg.env.frame_stack, cnn_keys, cfg.env.frame_stack_dilation)
 
+        if cfg.env.action_stack > 0 and "diambra" not in cfg.env.wrapper._target_:
+            env = ActionsAsObservationWrapper(env, cfg.env.action_stack, cfg.env.action_stack_dilation)
```
Comment (belerico) [inline sheeprl/utils/env.py:212]: In here is missing the noop action: the dilation is passed instead

Comment (michele-milesi) [reply sheeprl/utils/env.py:212]: Yeah, I am making the last changes and updating the docs

**Pull request merged**